bittensor/_dataset/dataset_impl.py

```diff
@@ -222,26 +323,52 @@ def get_directories(self, keys: list):
                     directory: Map{ Name: str, Hash: str, Size: int }: 
                         A random directory that lead to a datafile.
         """
+        def get_hashes(dataset_meta):
+
+            if self.IPFS_fails > 10:
```

Comment:
should have a global variable or something to indicate the number of fails we are willing to tolerate, it's best not to use an integer in an if statement, will be hard to find later :)